runtime/builtin_types.go

```diff
@@ -559,6 +560,33 @@ func builtinRange(f *Frame, args Args, kwargs KWArgs) (*Object, *BaseException)
 	return ListType.Call(f, []*Object{r}, nil)
 }
 
+func builtinRawInput(f *Frame, args Args, kwargs KWArgs) (*Object, *BaseException) {
+	if len(args) > 1 {
+		msg := fmt.Sprintf("[raw_]input expcted at most 1 arguments, got %d", len(args))
+		return nil, f.RaiseType(TypeErrorType, msg)
+	}
+
+	stdout := os.Stdout
+	stdin := NewFileFromFD(os.Stdin.Fd())
+	if stdin == nil {
+		msg := fmt.Sprintf("[raw_]input: lost sys.stdin")
+		return nil, f.RaiseType(RuntimeErrorType, msg)
+	}
+
+	if stdout == nil {
+		msg := fmt.Sprintf("[raw_]input: lost sys.stdout")
+		return nil, f.RaiseType(RuntimeErrorType, msg)
+	}
+
+	if len(args) == 1 {
+		pyPrint(f, args, "", "", stdout)
+	}
+
+	line, _ := stdin.readLine(MaxInt)
```

Comment:
I think I was wrong about implementing it this way :(

The trouble is that the C implementation uses fgets on the underlying C file pointer. fgets reads until it reaches `\n` but readLine will stop when it reaches `\r` and will convert that to a `\n`.

Probably the simplest thing to do here is to call `stdin.reader.ReadString("\n")` instead. This would mean you'd have to handle the EOF error case, however.